packages/waas/src/intents/transactions.ts

```diff
@@ -83,10 +122,54 @@ export function sendTransactions({
         value: ethers.BigNumber.from(tx.value || 0).toHexString(),
         data: ethers.utils.hexlify(tx.data || [])
       }
-    })
+    }),
+    transactionsFeeQuote
   })
 }
 
+function withTransactionFee(transactions: Transaction[], feeOption?: FeeOption): Transaction[] {
```

Comment:
This function will mutate the `transactions` array passed into `sendTransactions` - possibly unexpectedly by the caller. Is this expected?

Otherwise you could create a copy:
```
const extendedTxns = [...transactions]
// ...
extendedTxns.push(/* ... */)
// ...
return extendedTxns
```